### ISSUE TITLE
Notification: show build tip/warning messages on PR builds

### ISSUE THREAD

**Issue by humitos** [readthedocs/addons#88]
It would be really good to find a way to show build "tip" and "warning" notifications **on the rendered version of the documentation**. This will make people reading the PR version of that documentation to be aware of potential issues on that particular build.

We could:

- show a small 🔔 icon somewhere in the page that lists all the notification
- show a notification (similar to the current one) below/above the current one with the notification we want to communicate (the background color could 🟡 or 🟦 depending if it's a warning or a tip

### Example

This notification shown in the build's detail page (that most people don't see)

![Screenshot_2023-08-10_17-59-23](https://github.com/readthedocs/addons/assets/244656/29336887-72ec-4955-8642-cffd45e809a9)

could be shown **in the rendered version** as:

![Screenshot_2023-08-10_17-59-01](https://github.com/readthedocs/addons/assets/244656/4d368e16-d7b1-450a-b138-1a4937a5249a)

Related:
* https://github.com/readthedocs/readthedocs.org/issues/10598

**Comment by humitos** [comment]
> exposing a "build status" addon that just shows a little 🟢 🟡 🔴 status in the doc page

I was thinking that if it's _just the build status_, we could include an UI element inside the "This page was created from a pull request build" notification and keep the notification _and_ the status together in the same notification, since they are pretty related each other.

**Comment by ericholscher** [comment]
Yea, I think there's a few ways to show the data. I don't feel strongly about the UX, but a simple initial version would be neat just as a proof of concept, then we can build from there.

**Comment by humitos** [comment]
I started playing with this and found two interesting things:

We had a bug in our code for `builds.currents` when not sending the `url=` parameter. I opened a PR for this at https://github.com/readthedocs/readthedocs.org/pull/11068

> exposing a "build status" addon that just shows a little 🟢 🟡 🔴 status in the doc page.

To implement the use case that @ericholscher is describing here we don't have to cache the addons API responses for any page on PR builds. This is because the response will be _dynamic_ and will depend on the _ongoing_ build. This may not be a problem since we expect to have very little requests from PRs and we expect the cache to be invalidated frequently (on each push), but it's something to keep in mind if we decide to move forward with the implementation of your use case.

**Comment by ericholscher** [comment]
Good point on the caching. It might be a use case for an additional API call for that addon if it's enabled or something -- but I agree, it's probably fine to just not cache PR builds in this case since they are hit _way_ less often than production.

**Comment by humitos** [comment]
Now that we allowed anonymous access to the APIv3 (https://github.com/readthedocs/readthedocs.org/pull/11485), I think we should be able to do a POC for this. We should be able to hit https://docs.readthedocs.io/en/stable/api/v3.html#builds-listing and get the latest build and check the `finished` field of the first result.

We will need to add a `?version_slug=` query filter to the APIv3 endpoint first, tho.
